pennylane/devices/default_qubit.py

```diff
@@ -134,7 +134,7 @@ class DefaultQubit(QubitDevice):
         "QubitSum",
     }
 
-    observables = {"PauliX", "PauliY", "PauliZ", "Hadamard", "Hermitian", "Identity"}
+    observables = {"PauliX", "PauliY", "PauliZ", "Hadamard", "Hermitian", "Identity", "Projector"}
```

Comment:
@wongwsvincent, if you have a look at `QubitDevice`, the parent class, it _also_ has a set of supported observables:

https://github.com/PennyLaneAI/pennylane/blob/305279a39a1d70a49ff89b4d09644e62d65e735c/pennylane/_qubit_device.py#L112

This is because, even though `QubitDevice` is an abstract class to be inherited by simulator or hardware devices, simply inheriting it results in automatic support for the listed observables 🙂 This is partly what I alluded to in my comment in the issue --- by adding support for the `Projector` observerable to `QubitDevice`, then _any device that inherits from `QubitDevice`_ will also automatically support this new observable.

Reply:
@josh146 I actually tried adding "Projector" to the list of ```observables``` in _qubit_device.py like I did in default_qubit.py, but it didn't work when I tried it with a ```qiskit.aer``` device. 

After digging a bit more, I now realize that it was because ```observables``` is overwritten in the other package ```pennylane-qiskit```, which doesn't have the "Projector" included.  ```Projector``` is now working with ```qiskit.aer``` if I install the source code of pennylane-qiskit locally and implementing the same change in ```pennylane_qiskit/qiskit_device.py```. 

Does my understanding make sense? If so, I will create a new PR to commit the changes to ```pennylane_qiskit/qiskit_device.py```

Reply:
Extending this discussion to a ```cirq_device```, I find a [projector function in cirq](https://quantumai.google/reference/python/cirq/value/ProductState#projector). It's my first time looking into the cirq library, but adding ```"Projector": CirqOperation(lambda: cirq.ProductState.projector)``` to ```pennylane_cirq/cirq_device.py``` seems to do the job. I will submit a PR for this as well if it's what we want.

Reply:
> Does my understanding make sense? If so, I will create a new PR to commit the changes to pennylane_qiskit/qiskit_device.py

Yes this makes perfect sense! Thanks for investigating, it's shame it overwrites this list, otherwise it would have worked out of the box.

A PR contribution is always greatly appreciated 🙂 

> I find a projector function in cirq. It's my first time looking into the cirq library, but adding "Projector": CirqOperation(lambda: cirq.ProductState.projector) to the _observable_map in pennylane_cirq/cirq_device.py seems to do the job. I will submit a PR for this as well if it's what we want.

Yes, this is great! The general idea with the API is as follows:

- A generic, hardware compatible implementation is implemented in `QubitDevice`. This should allow _any_ inheriting device to automatically support these observables in a hardware-compatible manner.

- Some devices, however, might provide their own, more efficient implementations of these expectation values on simulators/hardware, like the Cirq devices. In which cases, these plugins should overwrite to directly use these direct methods 🙂

Reply:
Thanks for confirming, @josh146 !
I have created tickets for PR to pennylane-qiskit ([PR#137](https://github.com/PennyLaneAI/pennylane-qiskit/pull/137)) and pennylane-cirq ([PR#62](https://github.com/PennyLaneAI/pennylane-cirq/pull/62))

Reply:
Thanks @wongwsvincent! I have seen them come in, but will wait until this PR is merged before beginning the review process there :)